viam-cartographer/CMakeLists.txt

```diff
@@ -1,237 +1,80 @@
 cmake_minimum_required(VERSION 3.2)
 
 project(viam-cartographer)
-add_definitions(-DABSL_LEGACY_THREAD_ANNOTATIONS)
+set (CMAKE_CXX_STANDARD 17)
 
-set(VIAM_CARTOGRAPHER_MAJOR_VERSION 1)
-set(VIAM_CARTOGRAPHER_MINOR_VERSION 0)
-set(VIAM_CARTOGRAPHER_PATCH_VERSION 0)
-set(VIAM_CARTOGRAPHER_VERSION ${CARTOGRAPHER_MAJOR_VERSION}.${CARTOGRAPHER_MINOR_VERSION}.${CARTOGRAPHER_PATCH_VERSION})
-set(VIAM_CARTOGRAPHER_SOVERSION ${CARTOGRAPHER_MAJOR_VERSION}.${CARTOGRAPHER_MINOR_VERSION})
+if(DEFINED ENV{API_SOURCE_DIR})
```

Comment:
[q] where is this defined?

Reply:
It'd be something you set on the command line typically `API_SOURCE_DIR=/tmp/source/api make build` or with a -D option for cmake `cmake -DAPI_SOURCE_DIR=/tmp/source/api`

Also, I just moved this from lower down in the file (didn't create it) as I was rearranging things to a more logical flow.

Reply:
yeah we added this forever ago when protos still lived in rdk. Since the `build` step now depends on `buf` we can probably remove this at some point

Reply:
The reason you _might_ want to keep it (and why I left it) is when you're testing NEW api changes. This easily lets you use something local.